File: problem10/p10.rb
def primes n
  primes = []
  n.times do |num|
    primes.push(num) if is_prime? num
  end
  primes
end

def is_prime? n
  return false if n <= 1
  (2...n).each {|i| return false if n%i == 0}
  true
end

def list_sum list
  list.reduce(:+)
end

p list_sum(primes(10))

File: problem10/p10.js
function isPrime(num) {
  if (num <= 1) {
    return false;
  }
  var sqNum = Math.floor(Math.sqrt(num));
  while (sqNum >= 2) {
    if (num % sqNum === 0) {
      return false;
    }
    sqNum--;
  }
  return true;
}

function primes(num) {
  var primeList = [];

  for (var i = 0; i < num; i++) {
    if (isPrime(i)) {
      primeList.push(i);
    }
  }
  return primeList;
}

function sumList(list) {
  var sum = 0;
  for (var i = 0; i < list.length; i++) {
    sum += list[i];
  }
  return sum;
}

module.exports = {
  isPrime: isPrime,
  primes: primes,
  sumList: sumList
};
